web/cypress/integration/route/create-route-with-upstream.spec.js

```diff
@@ -0,0 +1,136 @@
+  };
+
+  beforeEach(() => {
+    // init login
+    cy.login();
+  });
+
+  it('should create route with upstream', () => {
+    // create a test upstream
+    cy.visit('/');
+    cy.contains('Upstream').click();
+    cy.wait(1000);
+    cy.contains('Create').click();
+
+    cy.get('#name').type(name.upstream);
+    cy.get('#desc').type('desc_by_autotest');
+    cy.get('#nodes_0_host').type('10.89.90.237');
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+
+    // go to route creat page
+    cy.visit('/');
+    cy.contains('Route').click();
+    cy.wait(1000);
+    cy.contains('Create').click();
+
+    // input name and description
+    cy.get('#name').type(name.route);
+    cy.contains('Next').click();
+
+    // select existed upstream_id will be disabled
+    cy.contains('Custom').click();
+    cy.contains(name.upstream).click();
+    cy.get(':input').should('be.disabled');
+    cy.wait(1000);
+
+    // select Custom upstream_id will not be disabled
+    cy.contains('test_upstream').click();
+    cy.contains('Custom').click();
+    cy.get(':input').should('not.be.disabled');
+
+    // change domain name/IP
+    cy.get('#nodes_0_host').clear().type('127.0.0.1');
+    cy.wait(1000);
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.contains('Goto List').click();
+    cy.url().should('contains', 'routes/list');
+  });
+
+  it('should edit this route with upstream', () => {
+    cy.visit('/');
+    cy.contains('Route').click();
+
+    // edit existed router
+    cy.get('[title=Name]').type(name.route);
+    cy.contains('Search').click();
+    cy.wait(1000);
+    cy.contains(name.route).siblings().contains('Edit').click();
+    cy.wait(1000);
+    cy.contains('Next').click();
+
+    // check if the changes have been saved
+    cy.get('#nodes_0_host').should('value', '127.0.0.1');
+
+    // select existed upstream_id will be disabled
+    cy.get('#upstream_id').click();
+    cy.contains(name.upstream).click();
+    cy.get(':input').should('be.disabled');
+    cy.wait(1000);
+
+    // select Custom upstream_id will not be disabled
+    cy.contains('test_upstream').click();
+    cy.contains('Custom').click();
+    cy.get(':input').should('not.be.disabled');
+
+    // change domain name/IP
+    cy.get('#nodes_0_host').clear().type('127.0.0.2');
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.contains('Goto List').click();
+    cy.url().should('contains', 'routes/list');
+
+    // check if the changes have been saved
+    cy.get('[title=Name]').type(name.route);
+    cy.contains('Search').click();
+    cy.wait(1000);
+    cy.contains(name.route).siblings().contains('Edit').click();
+    cy.wait(1000);
+    cy.contains('Next').click();
+    cy.get('#nodes_0_host').should('value', '127.0.0.2');
+  });
+
+  it('should delete ths test route and upstream', () => {
```

Comment:
`this`

Reply:
![image](https://user-images.githubusercontent.com/2106987/104830240-e4740400-58b7-11eb-8ae2-1f862af6214b.png)